apiserver/facades/client/application/application.go

```diff
@@ -329,74 +329,59 @@ func ConfigSchema() (configschema.Fields, schema.Defaults, error) {
 	return trustFields, trustDefaults, nil
 }
 
-func splitApplicationAndCharmConfig(inConfig map[string]string) (
-	appCfg map[string]interface{},
-	charmCfg map[string]string,
-	_ error,
-) {
-
-	providerSchema, _, err := ConfigSchema()
-	if err != nil {
-		return nil, nil, errors.Trace(err)
-	}
-	appConfigKeys := config.KnownConfigKeys(providerSchema)
-
-	appConfigAttrs := make(map[string]interface{})
-	charmConfig := make(map[string]string)
-	for k, v := range inConfig {
-		if appConfigKeys.Contains(k) {
-			appConfigAttrs[k] = v
-		} else {
-			charmConfig[k] = v
+func splitTrustFromApplicationConfig(cfg charm.Config) (bool, charm.Config) {
+	if trust, ok := cfg[coreapplication.TrustConfigOptionName]; ok {
+		delete(cfg, coreapplication.TrustConfigOptionName)
+		switch t := trust.(type) {
+		case bool:
+			return t, cfg
+		case string:
+			trustBool, err := strconv.ParseBool(t)
+			if err != nil {
+				return false, cfg
+			}
+			return trustBool, cfg
+		default:
+			return false, cfg
 		}
+	} else {
+		return false, cfg
 	}
-	return appConfigAttrs, charmConfig, nil
 }
 
-// splitApplicationAndCharmConfigFromYAML extracts app specific settings from a charm config YAML
-// and returns those app settings plus a YAML with just the charm settings left behind.
-func splitApplicationAndCharmConfigFromYAML(inYaml, appName string) (
-	appCfg map[string]interface{},
-	outYaml string,
+// splitTrustFromApplicationConfigFromYAML splits the trust flag from the application config
+// from the YAML string.
+//
+// The YAML config can be formatted in two ways:
+//  1. The YAML config is a set of key-value pairs corresponding to application
+//     config/settings (i.e. can include trust).
+//  2. The YAML config is a set of key-value pairs corresponding to application
+//     config/settings, indexed under the application name.
```

Comment:
This is a nice simplification.

We should re-instate the testing where possible.